lib/routes.js

```diff
@@ -39,17 +40,23 @@ module.exports = (
   })
 
   server.route({
-    method: 'POST',
+    method: ['GET', 'POST'],
     path: config.redirectUri,
     config: {
       auth: false
     },
     handler: async (request, h) => {
       try {
-        const { payload } = request
+        const { payload, method, query } = request
+
+        if (method === 'post' && config.passRequestToCacheMethods) {
```

Comment:
Can you explain in comments what you did in idm-tech? i.e. why it redirects to self?